Makefile

```diff
@@ -54,6 +54,12 @@ test: deps
 	@echo "=== $(INTEGRATION) === [ test ]: Running unit tests..."
 	@gocov test -race $(GO_FILES)... | gocov-xml > coverage.xml
 
+integration-test:
+	@echo "=== $(INTEGRATION) === [ test ]: running integration tests..."
+	@docker-compose -f tests/integration/docker-compose.yml up -d --build
+	@go test -v -tags=integration ./tests/integration/. || (ret=$$?; docker-compose -f tests/integration/docker-compose.yml down && exit $$ret)
+	@docker-compose -f tests/integration/docker-compose.yml down
```

Comment:
Perhaps we could rewrite this so `compose -f tests/integration/docker-compose.yml down` is only invoked once?

Like (untested):
```
	@go test -v -tags=integration ./tests/integration/. ; (ret=$$?; docker-compose -f tests/integration/docker-compose.yml down && exit $$ret)
```

Reply:
I've changed this line as you suggested because it works, I have also updated the project to use go modules